<html>
    <head>
        <script src="http://cdn.kendostatic.com/2014.3.1316/js/jquery.min.js"></script>
        <script src="C:\Users\aestrada\Documents\GitHub\JSDO\src\progress.util.js"></script>
        <script src="C:\Users\aestrada\Documents\GitHub\JSDO\src\progress.js"></script>
        <script src="C:\Users\aestrada\Documents\GitHub\JSDO\src\progress.session.js"></script>
        <script src="C:\Users\aestrada\Documents\GitHub\JSDO\src\progress.auth.js"></script>	
        
    </head>
    
    <body>
        <br/>
        <br/>
        <b>New Authentication Signature Tests</b>
        <br/>

        <script type="text/javascript">
            // These are tests for the story TP 39671 where there is no validation done
            // on the username and password passed to authenticate.

            var auth = new progress.data.AuthenticationProvider(
                "http://nbbedaestrada:8810/TS102/static/auth/j_spring_security_check"
                //"http://nbbedsrapuri1:8810/SSO_Token_Server/static/auth/j_spring_security_check"
            );
            
            // Call to authenticate with old signature
            // This doubles as an invalid type for username
            try {
                document.write("<br/><br>Invalid Type for userName Test Case: <br/>");
        
                auth.authenticate({ userName: "restuser", password: "password"})
                    .done(function (ap, result, info) {
                    console.log("authenticate method succeeded");
                
                })
                .fail(function (ap, result, info) {
                    console.log("authenticate method failed");
                });
                
                document.write("Failure: We didn't throw an error<br/>");
            } catch (e) {
                document.write("Success (Invalid type for userName): " + e);
            } 
                        
            // Call with invalid type for password
            try {
                document.write("<br/><br/>Invalid Type for password Test Case: <br/>");
                
                auth.authenticate("restuser", 1)
                    .done(function (ap, result, info) {
                    console.log("authenticate method succeeded");
                
                })
                .fail(function (ap, result, info) {
                    console.log("authenticate method failed");
                });
                
                document.write("Failure: We didn't throw an error<br/>");
            } catch (e) {
                document.write("Success (Invalid type for password): " + e);
            }
            
            // Call with undefined password
            try {
                document.write("<br/><br/>Undefined for password Test Case: <br/>");
                
                auth.authenticate("restuser")
                    .done(function (ap, result, info) {
                    console.log("authenticate method succeeded");
                
                })
                .fail(function (ap, result, info) {
                    console.log("authenticate method failed");
                });
                
                document.write("Failure: We didn't throw an error<br/>");
            } catch (e) {
                document.write("Success (Invalid type for password): " + e);
            }
            
            // Call with empty string for userName
            try {
                document.write("<br/><br/>Empty String for userName Test Case: <br/>");
                
                auth.authenticate("", "password")
                    .done(function (ap, result, info) {
                    console.log("authenticate method succeeded");
                
                })
                .fail(function (ap, result, info) {
                    console.log("authenticate method failed");
                });
                
                document.write("Failure: We didn't throw an error<br/>");
            } catch (e) {
                document.write("Success (Can't be empty string for userName): " + e);
            }

            // Call with empty string for password
            try {
                document.write("<br/><br/>Empty String for password Test Case: <br/>");
                
                auth.authenticate("restuser", "")
                    .done(function (ap, result, info) {
                    console.log("authenticate method succeeded");
                
                })
                .fail(function (ap, result, info) {
                    console.log("authenticate method failed");
                });
                
                document.write("Failure: We didn't throw an error<br/>");
            } catch (e) {
                document.write("Success (Can't be empty string for password): " + e);
            }
            
            // Call with empty string for password
            try {
                document.write("<br/><br/>Successful Authentication Test Case: <br/>");
                
                auth.authenticate("restuser", "password")
                    .done(function (ap, result, info) {
                    console.log("authenticate method succeeded");
                
                })
                .fail(function (ap, result, info) {
                    console.log("authenticate method failed");
                });
                
                document.write("Success: We didn't throw an error<br/>");
            } catch (e) {
                document.write("Failure: " + e);
            }
            
        </script> 
    </body>
</html>